python/ray/dataframe/dataframe.py

```diff
@@ -2308,9 +2308,43 @@ def mod(self, other, axis='columns', level=None, fill_value=None):
             "github.com/ray-project/ray.")
 
     def mode(self, axis=0, numeric_only=False):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        axis = pd.DataFrame()._get_axis_number(axis)
+
+        def mode_helper(df):
+            if numeric_only:
+                df = df.select_dtypes(exclude='object')
+            mode_df = df.mode(axis=axis)
+            return mode_df, mode_df.shape[axis]
+
+        def reindex_helper(df, *lengths):
+            max_len = max(lengths[0])
+            df = df.reindex(pd.RangeIndex(max_len), axis=axis)
+            return df
+
+        if axis == 0:
```

Comment:
You can probably reuse the code across both if cases, since they're largely the same

Reply:
resolved.